docs/source/platform/federation.mdx

```diff
@@ -0,0 +1,228 @@
+
+### At a glance
+
+The Graph Manager maintains both the [current state](https://www.apollographql.com/docs/platform/schema-registry/#using-the-schema-registry) of your graph as well as the [history](https://www.apollographql.com/docs/platform/schema-registry/#viewing-schema-change-history) of changes to that graph. When running GraphQL as a monolithic service with a static schema, the state of a data graph is well-represented by two components: operations and the schema. The schema represents what data is available in the graph, and operations, whether [registered](https://www.apollographql.com/docs/platform/operation-registry/#3-register-operations-from-your-client-bundle) or [collected dynamically](https://www.apollographql.com/docs/apollo-server/features/metrics/), represent the usage of that data graph.
+
+![Graph State](../img/federation/graph-current-state.png)
+
+The concept of a Federated Graph adds a single but important piece into the model for running and coordinating over a Graph: **One graph is composed of multiple implementing services**. This means that the state of the graph includes the set of services that implement it as well as _their_ states.
+
+The core concept of the Graph Manager in supporting federation is that each implementing service's partial schema should be tracked the same way as a monolithic data graph would be, and that the state of the Graph should be a reflection of the state of all of its dependencies.
+
+## Registering your implementing services
+
+If you're running a distributed GraphQL infrastructure, where implementing services [compose](https://www.apollographql.com/docs/apollo-server/federation/federation-spec/#fetch-service-capabilities) to form a complete schema, there's functionality to register each service's partial schema and still get all of the benefits of the schema registry.
+
+To push a single implementing service to Apollo, run `apollo service:push` with the `--serviceName` and `--endpoint` tags. The CLI will know where to fetch your local partial schema based on this endpoint. Note that running `apollo service:push` on any implementing service also registers the overall schema of the graph and writes files to GCS.
+
+As an example, running service push on a local “Accounts” implementing service might look like:
+
+```
+~$apollo service:push --serviceName="Accounts" --serviceURL="http://accounts-graphql:4001/ --endpoint="http://localhost:4001/"
+  ✔ Loading Apollo Project
+  ✔ Uploading service to Engine
+id      schema                        tag
+──────  ────────────────────────────  ───────
+190331  service-list-federation-demo  current
+```
+
+Pushing an implementing service to Apollo will update the [Services tab](#inspecting-your-graph).
```

Comment:
We should add a `>NOTE` or an additional sub-section here around handling failures in composition and how the implementing service will always be updated as long as its partial schema is "valid," but the graph will **not** be updated (i.e. the schema and the managed gateway config (also sidenote: let's align on the term "managed gateway config" instead of saying weird stuff about GCS everywhere))

Reply:
This also points out that if we're going to mention managed config here, we either need to bring that section higher in the document or just link downwards to it.

Reply:
We should also show an example output of what happens when service:push fails, and update this to have the new content that was added today in the messages

Reply:
EXAMPLE of what happens when you push a service and it composes fine and updates files. We should modify names and use the same example of services & graph across the board

GRAPH: space-explorer
VARIANT: current
SERVICES: users, launches, reservations

```
$ ../apollo-tooling/packages/apollo/bin/run service:push --serviceName registry --endpoint http://localhost:4001/api/graphql --engine https://engine-staging-graphql.apollographql.com/api/graphql --serviceURL http://engine-apigateway.svc.cluster.local:4001/api/graphql --tag staging
  ✔ Loading Apollo Project
  ✔ Loading Apollo Project
  ✔ Uploading service to Engine


The 'registry' service for the 'apollo-federated@staging' graph was updated

The gateway for the 'apollo-federated@staging' graph was updated with a new schema, composed from the updated 'registry' service

id      graph             variant
──────  ────────────────  ───────
da81e5  apollo-federated  staging

```